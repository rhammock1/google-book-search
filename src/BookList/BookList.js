import React from 'react';
import BookCard from './BookCard/BookCard';

const BookList = function(props) {

  return(
    <div className='book-list-container'>
      {/* Need to map through state.books to create book card for ea book */}
      {props.books.map((book) => {
        return(
          <BookCard book={book}/>
        )
      })}
      
    </div>
    )
}

BookList.defaultProps = {
  books:[]
}
export default BookList;