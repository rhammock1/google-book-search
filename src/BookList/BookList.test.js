import React, { createElement } from 'react';
import ReactDOM from 'react-dom';
import BookList from './BookList';
import renderer from 'react-test-renderer';


describe('Search component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BookList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer.create(<BookList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})