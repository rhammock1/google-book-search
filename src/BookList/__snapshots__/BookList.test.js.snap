// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders the UI as expected 1`] = `
<div
  className="book-list-container"
/>
`;
